### PR TITLE
FIX: correct label orientation for 0-90 deg nodes in plot_connectivity_circle

### PR DESCRIPTION
**Pull request by paavalipopov**
#### What does this implement/fix?

In [`mne.viz.circle._plot_connectivity_circle`](https://github.com/mne-tools/mne-python/blob/a6427d8e8b9e15b8aa63b50c4ffad20d390aa6af/mne/viz/circle.py#L130), the condition on line 341 `if angle_deg >= 270` that determines labels horizontal alignment only covers the 270–360° range (3–6 o'clock). Nodes in the 0–90° range (12–3 o'clock) are incorrectly treated as left-half nodes, and their labels appear pointing inward, which looks like a bug. 

Fix: extend the condition to `angle_deg >= 270` or `angle_deg < 90`, matching the full right half of the circle.

#### Additional information
Here's a visual comparison of connectograms generated using the original ("old") and patched ("new") versions of mne-tools. 

comparison

### CODE REVIEW

File: mne/viz/tests/test_connectivity_circle.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+
+def test_plot_connectivity_circle_label_orientation():
+    """Labels in the 0-90 deg polar range (12-3 o'clock) must not be flipped.
+
+    Regression test: previously the condition ``angle_deg >= 270`` missed the
+    [0, 90) range, incorrectly adding 180 degrees to those labels and setting
+    ha='right', which caused them to point inward instead of outward.
+    """
+    # 8 nodes → uniform angles: 0, 45, 90, 135, 180, 225, 270, 315 degrees.
```
Comment (CarinaFo) [inline mne/viz/tests/test_connectivity_circle.py]: Should be 9 nodes to be in line with the simulated nodes (l.22)

**Review by CarinaFo — COMMENTED** [review]
looks good to me, only a minor change in the test docs and the location of the test function (just updated)

File: mne/viz/tests/test_connectivity_circle.py (hunk 1 above)
Comment (paavalipopov) [reply mne/viz/tests/test_connectivity_circle.py]: Thanks for noticing this, I fixed it!

File: mne/viz/tests/test_connectivity_circle.py
Comment (CarinaFo) [inline mne/viz/tests/test_connectivity_circle.py:1]: general comment: could you add the test to test_cycle.py, I think we want to avoid having multiple test files for the cycle visualisation.

Comment (paavalipopov) [reply mne/viz/tests/test_connectivity_circle.py:1]: I merged this test into test_cycle.py.

**Comment by CarinaFo** [comment]
It looks like this branch includes several unrelated commits or working-tree changes. Could you reset/rebase the branch onto current main and recommit just the relevant files? Based on the commit history, three commits are unrelated to the current work

**Comment by scott-huberty** [comment]
Hey @paavalipopov Can you please create a CircleCI account by signing into CircleCI via GitHub? After that you can push a commit here, and the documentation build job should run.

**Review by CarinaFo — APPROVED** [review]
Thank you and congrats to your first PR, looks good to me.

**Comment by paavalipopov** [comment]
Thanks for the review and the help with the cleanup, @CarinaFo! Glad to contribute to mne tools.

File: mne/viz/tests/test_circle.py [hunk 2]
```diff
@@ -33,3 +36,53 @@ def test_plot_channel_labels_circle():
...
+    Regression test: previously the condition ``angle_deg >= 270`` missed the
+    [0, 90) range, incorrectly adding 180 degrees to those labels and setting
+    ha='right', which caused them to point inward instead of outward.
+    """
+    # 9 nodes → uniform angles: 0, 40, 80, 120, 160, 200, 240, 280, 320 degrees.
+    # This guarantees coverage of all four quadrants, including the previously
+    # broken 0–90 range (nodes n0 at 0°, n1 at 40°, and n2 at 80°).
+    n_nodes = 9
+    rng = np.random.default_rng(0)
+    con = rng.uniform(0, 1, size=(n_nodes, n_nodes))
```
Comment (drammock) [inline mne/viz/tests/test_circle.py]: no need to bother with randomness in a test that only cares about label orientation
```suggestion
    con = np.ones((n_nodes, n_nodes))
```

Comment (drammock) [reply mne/viz/tests/test_circle.py]: could also do `1 - np.eye(n_nodes)` and avoid the `fill diagonal` but that seems a bit too clever, might puzzle a future maintainer

File: mne/viz/tests/test_circle.py [hunk 3]
```diff
@@ -33,3 +36,53 @@ def test_plot_channel_labels_circle():
...
+
+    # node_angles defaults to np.linspace(0, 2*pi, n_nodes, endpoint=False)
+    angles_deg = np.linspace(0, 360, n_nodes, endpoint=False)
+
+    assert len(label_texts) == n_nodes, (
+        f"Expected {n_nodes} label texts, found {len(label_texts)}"
+    )
+
+    for i, name in enumerate(node_names):
+        angle = angles_deg[i]
```
Comment (drammock) [inline mne/viz/tests/test_circle.py]:
```suggestion
    for angle, name in zip(angles_deg, node_names, strict=True):
```

File: mne/viz/tests/test_circle.py [hunk 4]
```diff
@@ -33,3 +36,53 @@ def test_plot_channel_labels_circle():
...
+                f"ha='left', got '{ha}'"
+            )
+        else:
+            # Left half: text is flipped 180° so it stays upright; ha='right'
+            # anchors the right edge at the node, text extends leftward/outward.
+            assert ha == "right", (
+                f"Node '{name}' at {angle:.1f}° (left half) should have "
+                f"ha='right', got '{ha}'"
+            )
+    plt.close(fig)
```
Comment (drammock) [inline mne/viz/tests/test_circle.py]: We have a fixture that makes this happen automatically

```suggestion
```

**Review by drammock — APPROVED** [review]
thanks @paavalipopov this is a nice PR. Thanks for the thorough test. I have a couple of suggestions to tidy it a bit, which I think should be uncontroversial so I'll apply them and mark for merge.

File: mne/viz/tests/test_circle.py (hunk 2 above)
Comment (CarinaFo) [reply mne/viz/tests/test_circle.py]: this breaks the _plot_connectivity_circle function (division by zero error in the line where the connectivity values get normalised). From what I understand, this is used to map the colors according to connectivity strength and should probably be addressed in a new PR as it is a edge case within plot_connectivity_circle.

Comment (drammock) [reply mne/viz/tests/test_circle.py]: Oops. PR is not merged yet so revert this to using random values I guess

Comment (paavalipopov) [reply mne/viz/tests/test_circle.py]: I modified the values in the connectivity matrix to add some variability without using rng, it works now.
```python
    con = np.ones((n_nodes, n_nodes))
    con[::2, ::2] = 0  # add some zeros to avoid 0-div in normalization
```

**Comment by paavalipopov** [comment]
It seems like build_docs check is failing due to a warning in the [mne.viz.ui_events.disable_ui_events](https://github.com/mne-tools/mne-python/blob/main/mne/viz/ui_events.py) docstring. I did not touch this function, is there anything I should do on my end to fix it?

**Comment by drammock** [comment]
CircleCI failure is being fixed in #13868, will merge manually once other CIs come back green
